src/mscorlib/src/System/Guid.cs

```diff
@@ -564,7 +564,17 @@ private static bool TryParseGuid(String g, GuidStyles flags, ref GuidResult resu
                 }
 
                 // Read in the number
-                uint number = (uint)Convert.ToInt32(guidString.Substring(numStart, numLen),16);
+                uint number;
+                try
+                {
+                    number = (uint)Convert.ToInt32(guidString.Substring(numStart, numLen), 16);
```

Comment:
Presumably this is the behavior that the StringToInt method is providing?  Seems like once there's a helper it should be consistently called...